idea/idea-frontend-api/src/org/jetbrains/kotlin/idea/frontend/api/calls/KtCall.kt

```diff
@@ -53,12 +55,48 @@ public class KtVariableWithInvokeFunctionCall(
     override val targetFunction: KtCallTarget
 ) : KtDeclaredFunctionCall()
 
+/**
+ * Represents a direct function call with arguments
+ */
+public sealed class KtCallWithArguments : KtDeclaredFunctionCall() {
+    public abstract val argumentMapping: LinkedHashMap<KtValueArgument, KtValueParameterSymbol>
```

Comment:
W00t! I definitely need this new abstraction! https://github.com/JetBrains/intellij-kotlin/pull/81/commits/e34aecbcd966f0688b435f85117195543b7b51a1#diff-41b037ea4d3673022f43bfaeac2b249659baa8069275309ff27c7649ca07d394

Reply:
Also all of the following differentiated abstractions should be helpful for commonizing function call and annotations, thanks!